src/utils/dateHelpers.js

```diff
@@ -0,0 +1,32 @@
+export const printDate = (dateString) => {
+  const date = new Date(dateString);
+  if (isNaN(date)) {
+    return 'Invalid Date';
+  }
+  const month = date.getUTCMonth() + 1;
+  const day = date.getUTCDate();
+
+  return `${date.getUTCFullYear()}-${month < 10 ? `0${month}` : month}-${
+    day < 10 ? `0${day}` : day
+  }`;
+};
+
+export const sortActivationKeys = (
+  activationKeys,
+  sortColumnIndex,
+  sortDirection,
+  columnsMap
+) => {
+  return [...activationKeys].sort((a, b) => {
+    const aValue = a[columnsMap[sortColumnIndex]];
+    const bValue = b[columnsMap[sortColumnIndex]];
+    if (sortColumnIndex === 4) {
+      const aDate = new Date(a.updatedAt);
+      const bDate = new Date(b.updatedAt);
+      return sortDirection === 'asc' ? aDate - bDate : bDate - aDate;
+    }
+    if (aValue < bValue) return sortDirection === 'asc' ? -1 : 1;
+    if (aValue > bValue) return sortDirection === 'asc' ? 1 : -1;
+    return 0;
+  });
+};
```

Comment:
Thought [non-blocking]: Looking at this function some more, I thought it should be in dateHelpers because I thought it was _only_ sorting by date. It looks like this supports some other sorting too.

I'm not sure what is better: 
* leaving this inline in the component where you originally had it
* updating this function to only sort by date (that is the only sorting we do on the activation keys table as of now, right?)
* moving this to another util file for activation keys helper methods

I'll leave that decision to you, but I think its worth considering if we should do something like that. Maybe this function also needs a comment, because it is pretty unclear (see my confusion at the beginning lol) what this function is doing.

Thoughts?

Reply:
Thought [non-blocking]: I also am not too fond of the correlation between sortIndex and the behavior of this function. That tightly couples this function to the way that table is defined.

Reply:
> Thought [non-blocking]: Looking at this function some more, I thought it should be in dateHelpers because I thought it was _only_ sorting by date. It looks like this supports some other sorting too.
> 
> I'm not sure what is better:
> 
> * leaving this inline in the component where you originally had it
> * updating this function to only sort by date (that is the only sorting we do on the activation keys table as of now, right?)
> * moving this to another util file for activation keys helper methods
> 
> I'll leave that decision to you, but I think its worth considering if we should do something like that. Maybe this function also needs a comment, because it is pretty unclear (see my confusion at the beginning lol) what this function is doing.
> 
> Thoughts?

I am going to leave this with dateHelper utils and reconfigure the function to be more applicable to date sorting functionality _only_ for clarity reasons. So I opted to chose the second option.

Reply:
> Thought [non-blocking]: I also am not too fond of the correlation between sortIndex and the behavior of this function. That tightly couples this function to the way that table is defined.

I do understand your concerns so there is no overlap i have removed the index reference , and just kept this function strictly to date sorting.